### PR TITLE
Add blueoil.io module for I/O with GCS support

### PR DESCRIPTION
**Pull request by hadusam**
## What this patch does to fix the issue.
A part of #1205

### CODE REVIEW

**Review by iizukak — APPROVED** [review]
OA

File: blueoil/io/file_io.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import numpy as np
+from PIL import Image
+from tensorflow.io import gfile
+
+
+def _is_local_path(path: str) -> bool:
+    for p in {"gs://", "s3://", "hdfs://"}:
+        if path.startswith(p):
+            return False
+    return True
```
Comment (tfujiwar) [inline blueoil/io/file_io.py:28]: You can pass a tuple to `startswith`.
https://docs.python.org/3/library/stdtypes.html#str.startswith

File: tests/unit/io_tests/test_file_io.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+def test_mkdir():
+    with tempfile.TemporaryDirectory() as d:
+        path = os.path.join(d, "test")
+        file_io.mkdir(path)
+        assert os.path.isdir(path)
+
+
+def test_makedirs():
+    with tempfile.TemporaryDirectory() as d:
+        path = os.path.join(os.path.join(d, "test"), "test")
```
Comment (tfujiwar) [inline tests/unit/io_tests/test_file_io.py:61]: Is this enough?
```suggestion
        path = os.path.join(d, "test", "test")
```

**Review by tfujiwar — COMMENTED** [review]
Just small suggestions.

File: tests/unit/io_tests/test_file_io.py (hunk 2 above)
Comment (hadusam) [reply tests/unit/io_tests/test_file_io.py]: Fixed! 👍

File: blueoil/io/file_io.py (hunk 1 above)
Comment (hadusam) [reply blueoil/io/file_io.py]: Oh, thanks! I simplify checking the local path and remove this function. Please check it! 👍

**Review by tfujiwar — APPROVED** [review]
RA
Thank you~~

**Pull request merged**